deebot_client/map.py

```diff
@@ -314,74 +480,114 @@ def get_base64_map(self, width: int | None = None) -> bytes:
             and width == self._last_image.width
             and not self._map_data.changed
         ):
-            _LOGGER.debug("[get_base64_map] No need to update")
-            return self._last_image.base64_image
+            _LOGGER.debug("[get_svg_map] No need to update")
+            return self._last_image.svg_image
 
-        _LOGGER.debug("[get_base64_map] Begin")
-        image = Image.new("RGBA", (6400, 6400))
-        draw = DashedImageDraw(image)
+        _LOGGER.debug("[get_svg_map] Begin")
 
+        image = Image.new("RGBA", (6400, 6400))
+        draw = ImageDraw.ImageDraw(image)
         self._draw_map_pieces(draw)
-
-        # Draw Trace Route
-        if len(self._map_data.trace_values) > 0:
-            _LOGGER.debug("[get_base64_map] Draw Trace")
-            draw.line(self._map_data.trace_values, fill=_COLORS[_TRACE_MAP], width=1)
+        del draw
 
         image_box = image.getbbox()
-        for subset in self._map_data.map_subsets.values():
-            _draw_subset(subset, draw, image_box)
 
-        del draw
+        if image_box:
+            image_box_center = (
+                (image_box[0] + image_box[2]) / 2,
+                (image_box[1] + image_box[3]) / 2,
+            )
 
-        _draw_positions(self._map_data.positions, image, image_box)
+            _LOGGER.debug("[get_svg_map] Crop Image")
+            cropped = image.crop(image_box)
+            del image
 
-        _LOGGER.debug("[get_base64_map] Crop Image")
-        cropped = image.crop(image_box)
-        del image
+            _LOGGER.debug(
+                "[get_svg_map] Map current Size: X: %d Y: %d",
+                cropped.size[0],
+                cropped.size[1],
+            )
 
-        _LOGGER.debug("[get_base64_map] Flipping Image")
-        cropped = ImageOps.flip(cropped)
+            _LOGGER.debug("[get_svg_map] Saving to buffer")
+            buffered = BytesIO()
+            cropped.save(buffered, format="PNG")
+            del cropped
 
-        _LOGGER.debug(
-            "[get_base64_map] Map current Size: X: %d Y: %d",
-            cropped.size[0],
-            cropped.size[1],
-        )
+            base64_bg = base64.b64encode(buffered.getvalue())
 
-        new_size = None
-        if width is not None and width > 0:
-            height = int((width / cropped.size[0]) * cropped.size[1])
-            _LOGGER.debug(
-                "[get_base64_map] Resize based on the requested width: %d and calculated height %d",
-                width,
-                height,
+            # Build the SVG XML
+
+            svg_positions = _get_svg_positions(self._map_data.positions, image_box)
+
+            svg_subset_elements: list[svg.Element] = [
+                _get_svg_subset(subset, image_box)
+                for subset in self._map_data.map_subsets.values()
+            ]
+
+            svg_rooms_elements, svg_rooms_labels = self._get_svg_rooms(
```

Comment:
I was taking a look at other PRs, and yeah, as far as I understand, rooms currently come from two sources: the map background itself (available only on newer models, and currently not supported by the library causing https://github.com/DeebotUniverse/Deebot-4-Home-Assistant/issues/268 ), and as discrete map sub sets (available since older models, and that was how I was rendering rooms ina  custom extension for my old, and now broken, Ozmo 900, from which I borrowed the code sent in the PR). 

Anyway, ok, I will remove the room rendering part from the PR.

Reply:
https://github.com/DeebotUniverse/Deebot-4-Home-Assistant/issues/268 will be fixed with https://github.com/DeebotUniverse/Deebot-4-Home-Assistant/issues/268

The map background of newer models supports additional types and the PR will add colors for them

Reply:
> Anyway, ok, I will remove the room rendering part from the PR.

You can create a new branch out of this one. When this PR is merged only rebase it and create a new PR and we can talk about adding colorized rooms. Still not sure if we should add the names as the user wants to ave them in their language and not in english

Reply:
Room rendering code removed. Let me know if I should perform a rebase with squash, or if you will take care of it. Meanwhile the room rendering code has been parked on a new branch.